sys/include/net/sock/dtls.h

```diff
@@ -0,0 +1,556 @@
+ * we could alternatively set the `timeout` parameter of the function to a value
+ * != @ref SOCK_NO_TIMEOUT. If an error occur we just ignore it and continue looping.
+ * We can reply to an incoming message using its `session`.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * while (1) {
+ *     ssize_t res;
+ *     sock_dtls_session_t session;
+ *
+ *     res = sock_dtls_recv(&dtls_sock, &session, buf, sizeof(buf)),
+ *                          SOCK_NO_TIMEOUT);
+ *     if (res > 0) {
+ *         puts("Received a message");
+*         if (sock_dtls_send(&dtls_sock, buf, res, &session) < 0) {
+ *             puts("Error sending reply");
+ *         }
+ *     }
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * ### Client Operation
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * #include "net/sock/udp.h"
+ * #include "net/sock/dtls.h"
+ * #include "net/ipv6/addr.h"
+ * #include "net/credman.h"
+ *
+ * int main(void)
+ * {
+ *     sock_udp_t udp_sock;
+ *     sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ *     local.port = 12345;
+ *     sock_udp_create(&udp_sock, &local_ep, NULL, 0);
+ *
+ *     sock_udp_ep_t remote;
+ *     remote.port = DTLS_DEFAULT_PORT;
+ *     remote.netif = gnrc_netif_iter(NULL)->pid;   // only if gnrc_netif_numoff == 1
+ *
+ *     if (!ipv6_addr_from_str((ipv6_addr_t *)remote.addr.ipv6, addr_str)) {
+ *         puts("Error parsing destination address");
+ *         return;
+ *     }
+ *
+ *     res = sock_dtls_create(&dtls_sock, &udp_sock, DTLS_SOCK_CLIENT_TAG, 0);
+ *     if (res < 0) {
+ *         puts("Error creating DTLS sock");
+ *         return;
+ *     }
+ *
+ *     res = sock_dtls_establish_session(&dtls_sock, &remote, &session);
+ *     if (res < 0) {
+ *         printf("Error establishing session: %d\n", (int)res);
+ *         goto end;
+ *     }
+ *
+ *     res = sock_dtls_send(&dtls_sock, &session, data, datalen);
+ *     if (res < 0) {
+ *         printf("Error sending DTLS message: %d\n", (int)res);
+ *         goto end;
+ *     }
+ *
+ *     res = sock_dtls_recv(&dtls_sock, &session, rcv, sizeof(rcv), SOCK_NO_TIMEOUT);
+ *     if (res < 0) {
+ *         printf("Error receiving DTLS message: %d\n", (int)res);
+ *         goto end;
+ *     }
+ *
+ *     end:
+ *         puts("Terminating");
+ *         sock_dtls_terminate_session(&dtls_sock, &session);
+ *         sock_dtls_close(&dtls_sock);
+ *  }
+ * }
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ * This is an example of a DTLS echo client.
+ *
+ * Like the server, we must first create the UDP sock.
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ {.c}
+ * sock_udp_t udp_sock;
+ * sock_udp_ep_t local = SOCK_IPV6_EP_ANY;
+ * local.port = 12345;
+ * sock_udp_create(&udp_sock, &local_ep, NULL, 0);
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ *
+ * After that, we set the address of the remote endpoint where the packet is
+ * supposed to be sent to and the port the server is listening to which is
+ * DTLS_DEFAULT_PORT(20220).
```

Comment:
```suggestion
 * DTLS_DEFAULT_PORT (20220).
```

(otherwise it looks like a macro invocation)

Reply:
Fixed.